frontend/packages/erd-core/src/components/ERDRenderer/ERDContent/ERDContent.tsx

```diff
@@ -95,41 +95,25 @@ export const ERDContentInner: FC<Props> = ({
   const {
     state: { loading },
   } = useERDContentContext()
+  // TODO: remove activeNodeId state
   const [activeNodeId, setActiveNodeId] = useState<string | null>(null)
 
   useUpdateNodeCardinalities(nodes, relationships, setNodes)
   useInitialAutoLayout()
   useFitViewWhenActiveTableChange(
     enabledFeatures?.fitViewWhenActiveTableChange ?? true,
   )
+  useSyncHighlightsActiveTableChange()
 
-  const handleNodeClick = useCallback(
-    (nodeId: string) => {
-      setActiveNodeId(nodeId)
-      updateActiveTableName(nodeId)
-
-      const { nodes: updatedNodes, edges: updatedEdges } =
-        highlightNodesAndEdges(nodes, edges, nodeId)
-
-      setEdges(updatedEdges)
-      setNodes(updatedNodes)
-    },
-    [edges, nodes, setNodes, setEdges],
-  )
```

Comment:
The reason was that Active highlighting was only done on click.
Monitoring active state and implementing it with useEffect, so that it is highlighted in all patterns.